### PR TITLE
Remove remote EContactTypeahead usages

### PR DESCRIPTION
**Pull request by leogrim**
@stkem @raymondng @eishay

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/typeahead/abook/EContactTypeahead.scala [hunk 1]
```diff
@@ -81,6 +81,7 @@ object EContactTypeaheadBase {
   }
 }
 
+// todo(Léo) ** Not In Use ** - Remove if the performance hit from encapsulating EContactTypeahead in ABook does not prove problematic
 // "Remote"; uses abookServiceClient
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/typeahead/abook/EContactTypeahead.scala:85]: Cc @raymondng @stkem @eishay 
The code is still here until for now.

**Comment by eishay** [comment]
what are the implications on the architecture level?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala [hunk 2]
```diff
@@ -62,18 +60,19 @@ class TypeaheadCommander @Inject()(
   private def emailId(email: EmailAddress) = s"email/${email.address}"
   private def socialId(sci: SocialUserBasicInfo) = s"${sci.networkType}/${sci.socialId.id}"
 
-  def queryContacts(userId: Id[User], search: Option[String], limit: Int): Future[Seq[EContact]] = {
-    abookServiceClient.prefixQuery(userId, limit, search, None)
+  def queryContacts(userId: Id[User], search: Option[String], limit: Int): Future[Seq[RichContact]] = {
+    search match {
+      case Some(query) => abookServiceClient.contactTypeahead(userId, query, Some(limit)).map { hits => hits.map(_.info) }
+      case None => Future.successful(Seq.empty) //todo(Léo): check with mobile about this
+    }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala:67]: @davoda @jaredpetker @ebfio @2is10 
This is used by a bunch of endpoints, could you update me on their status?

GET     /m/1/user/connections/all
GET     /m/1/user/connections/social  
GET     /m/1/user/connections/email

GET     /ext/contacts
GET     /ext/nonusers

GET    /site/user/connections/all/search

**Comment by leogrim** [comment]
@eishay 
We go to ABook directly for EContact typeahead, which thereby gets encapsulated in ABook, together with EContact (final touch on the way once we've migrated requesting services to the new endpoints). 

That should also alleviate some load on Shoebox, besides getting stuff out of it in general. It's fairly easy to reverse in TypeaheadCommander if there is a real performance bottleneck, but I don't expect so. We are doing much more work for User typeahead as we go to Search for instance.

**Comment by eishay** [comment]
Good, thanks. I assume that we won't open abook as an endpoint and still route the calls through the shoebox, right?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala (hunk 2 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala:67]: `/ext/nonusers` just returns an empty JS array for backwards compatibility
`/ext/contacts` is used for To: box suggestions

**Comment by leogrim** [comment]
Yes, hence some discussion before going ahead with this to determine whether doing the trip to ABook  made sense. Given that we are also doing a number of queries to Memcache and that it moves network / processing load away from Shoebox, decided that it was worth trying. 

As a side product, having EContact in ABook will enable me to have a better foreign key reference from EContact to EmailAccount for better integrity guarantees there. I have the new Contact interning logic in a separate branch, to be merged once I can add in that reference.

**Comment by eishay** [comment]
nice, thanks!

**Comment by leogrim** [comment]
@eishay It's in production, no noticeable performance problem, if anything it actually feels faster (but I'm biased ;)

**Pull request merged**